pkgs/tools/misc/iay/default.nix

```diff
@@ -0,0 +1,30 @@
+{ lib
+, rustPlatform
+, fetchCrate
+, stdenv
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "iay";
+  version = "0.4.0";
+
+  src = fetchCrate {
```

Comment:
There is a cargo lock in the repo https://github.com/aaqaishtyaq/iay/blob/trunk/Cargo.lock . Why not fetch that?

Reply:
Thank you for reviewing!

So, should fetchCrate be avoided if the repository has Cargo.lock? It is the first time I package a Rust program and https://nixos.org/manual/nixpkgs/stable/#rust does not mention whether fetchCrate is discouraged or not. Would you point out guidelines, if you are aware of any?